airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -170,6 +172,15 @@ static void consumeWriteStream(final AirbyteMessageConsumer consumer) throws Exc
           LOGGER.error("Received invalid message: " + inputString);
         }
       }
+    } finally {
+      for (final Thread runningThread : ThreadUtils.getAllThreads()) {
+        if (!runningThread.getName().equals(currentThread.getName())
+            && runningThread.getThreadGroup().getName().equals(currentThread.getThreadGroup().getName())
+            && !runningThread.isDaemon()) {
+          LOGGER.warn("A child non-daemon thread is still active {}", runningThread);
+          // Should we call runningThread.interrupt(); ?
```

Comment:
on should we force a system exit?

Reply:
I guess that's even more brutal than my `runningThread.interrupt();` 

but @subodh1810 feels it's safer to just print a log for the moment before doing anything to interrupt/kill that may have bad side unexpected effects for the moment.

(At least, we would get some traces in the logs even though we wouldn't solve anything in particular here)

Reply:
I modified the PR to include a system.exit as last resort